### ISSUE TITLE
toc-validation (MD073) does not correctly consider indentation

### ISSUE THREAD

**Issue by chrisgrieser** [rvben/rumdl#353]
### rumdl version

0.1.4

### Installation method

Pre-built binary

### Bug description

Two bugs, which I presume to be related to each other, thus creating one issue for them.

- Bug 1: indentation is not linted
- Bug 2: initially, newly created etries via formatting are wrongly indented. On a second run, the indentation is correct

### Steps to reproduce

Bug 1:

```md
<!-- toc -->
- [Hello](#hello)
- [Another](#another)
- [Heading](#heading)
<!-- tocstop -->

## Hello

### Another

## Heading
```

The second heading is level-3, thus requiring indentation. This is incorrect in the ToC, but not reported.

---

Bug 2:

Given this file:
```md
<!-- toc -->
<!-- tocstop -->

## Hello

### Another

## Heading
```

Now running formatting results in this in the second heading being only indented with 2 spaces, not 4 as specified in my config:

<img width="1335" height="496" alt="Image" src="https://github.com/user-attachments/assets/a6b9fde2-4c83-4c0a-b337-4bc45cd69f4b" />

However, by running formatting a second time, everything is correct.

### Expected behavior

see above

### Actual behavior

see above

### Configuration

```toml
[toc-validation]
enabled = true
```

### Operating system

macOS

**Comment by rvben** [comment]
Thanks for reporting @chrisgrieser. Will look at it.

**Comment by rvben** [comment]
Fixed in v0.1.5. Thanks for the detailed report @chrisgrieser!

TOC entry indentation is now validated against heading levels (H2 = 2-space indent, H3 = 4-space indent, etc.).

**Comment by chrisgrieser** [comment]
> H2 = 2-space indent, H3 = 4-space indent, etc.

But depending on the ul-indent config, it should be 4/8 spaces though?

**Comment by rvben** [comment]
Fixed in cd8ca84f.

MD073 now reads the `indent` value from MD007 by default, so your TOC will use 4-space indentation if you have:

```toml
[MD007]
indent = 4
```

You can also override it explicitly in MD073 if needed:

```toml
[MD073]
enabled = true
indent = 4  # Override MD007's value
```

Documentation updated at [docs/md073.md](https://rumdl.dev/md073#indentation-configuration).

Thanks for the report @chrisgrieser!
